File: Array_String/8-Rotate-String/Cpp/isRotation.cpp
#include <iostream>
using namespace std;

bool isSubstring(string str1, string str2) {
    // Check if str2 is a substring of str1
    return str1.find(str2) != string::npos;
}

bool isRotation(string str1, string str2) {
    int len = str1.length();
    /* Check that str1 and str2 are equal length and not empty */
    if (len == str2.length() && len > 0) {
        /* Concatenate str1 and str1 within new buffer */
        string str1str1 = str1 + str1;
        return isSubstring(str1str1, str2);
    }
    return false;
}

int main() {
    string s1 = "waterbottle";
    string s2 = "erbottlewat";

    if (isRotation(s1, s2)) {
        cout << s2 << " is a rotation of " << s1 << endl;
    }
    else {
        cout << s2 << " is not a rotation of " << s1 << endl;
    }

    return 0;
}

File: Array_String/8-Rotate-String/Python/isRotation.py
def isSubstring(s1, s2):
    return s2 in s1

def isRotation(s1, s2):
    length = len(s1)
    # Check that s1 and s2 are equal length and not empty
    if length == len(s2) and length > 0:
        # Concatenate s1 and s1 within new buffer
        s1s1 = s1 + s1
        return isSubstring(s1s1, s2)
    return False

def main():
    s1 = "waterbottle"
    s2 = "erbottlewat"
    print(isRotation(s1, s2))  # Output: True

if __name__ == "__main__":
    main()